e2e/components/DataTable/DataTable-test.avt.e2e.js

```diff
@@ -0,0 +1,373 @@
+        id: 'components-datatable-dynamic--default',
+        globals: {
+          theme: 'white',
+        },
+      });
+      await expect(page).toHaveNoACViolations(
+        'components-datatable-dynamic--default'
+      );
+    });
+
+    test('default keyboard navigation sequence', async ({ page }) => {
+      await visitStory(page, {
+        component: 'DataTable',
+        id: 'components-datatable-dynamic--default',
+        globals: {
+          theme: 'white',
+        },
+      });
+
+      await expect(page).toHaveNoACViolations(
+        'components-datatable-dynamic--default'
+      );
+
+      // Start off by manually focusing the search input
+      await page.getByRole('searchbox').focus();
+      await expect(page.getByRole('searchbox')).toBeFocused();
+
+      // Navigate to the gear/settings button
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('button', { name: 'Settings' })
+      ).toBeFocused();
+
+      // Navigate to the select all checkbox
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).toBeFocused();
+
+      // Pressing enter on the select all checkbox shouldn't do anything
+      await page.keyboard.press('Enter');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).not.toBeChecked();
+
+      // Pressing space should check the select all checkbox
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).toBeChecked();
+
+      // Every checkbox should be checked
+      for (const checkbox of await page.getByRole('checkbox').all()) {
+        await expect(checkbox).toBeChecked();
+      }
+
+      // Pressing space should uncheck the select all checkbox
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).not.toBeChecked();
+      // Every checkbox should no longer be checked
+      for (const checkbox of await page.getByRole('checkbox').all()) {
+        await expect(checkbox).not.toBeChecked();
+      }
+
+      // Navigate to the first expansion button
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('button', { name: 'Expand current row' }).first()
+      ).toBeFocused();
+      // Expand the first row
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('heading', { name: 'Expandable row content' }).first()
+      ).toBeVisible();
+
+      // Navigate to the first row selection checkbox and check it
+      await page.keyboard.press('Tab');
+      await page.keyboard.press('Space');
+      await expect(page.getByText('1 item selected')).toBeVisible();
+
+      // Navigate backwards up into the batch action bar
+      await page.keyboard.press('Shift+Tab');
+      await page.keyboard.press('Shift+Tab');
```

Comment:
Seems like two of these tab stops are required because the Search and Settings are focusable even when the batch action toolbar is active. Would it be worth spinning this into a separate issue to toggle `aria-hidden` and a `tabIndex` change based on `batchActionProps.shouldShowBatchActions`?

I was able to remove 1 tab stop by passing in a custom `tabIndex` to `TableToolbarSearch` and `TableToolbarMenu` in the `Dynamic` story like so:

```jsx
<TableToolbarMenu
   aria-hidden={ batchActionProps.shouldShowBatchActions ? true : false }
    tabIndex={ batchActionProps.shouldShowBatchActions ? -1 : 0 } />
```

But it doesn't seem to be passing properly to `TableToolbarSearch`. Small issue, but just wanted to note that

Reply:
Going to take care of this one in https://github.com/carbon-design-system/carbon/pull/14386